stubs/python-jenkins/jenkins/__init__.pyi

```diff
@@ -0,0 +1,220 @@
+    def get_build_test_report(self, name: str, number: int, depth: int = 0) -> Mapping[Any, Any] | None: ...
+    def get_build_artifact(self, name: str, number: int, artifact: str) -> Mapping[Any, Any]: ...
+    def get_build_artifact_as_bytes(self, name: str, number: int, artifact: str) -> bytes: ...
+    def get_build_stages(self, name: str, number: int) -> Mapping[Any, Any]: ...
+    def get_queue_info(self) -> Mapping[Any, Any]: ...
+    def cancel_queue(self, id: int) -> None: ...
+    def get_info(self, item: str = "", query: str | None = None) -> Mapping[Any, Any]: ...
+    def get_whoami(self, depth: int = 0) -> Mapping[Any, Any]: ...
+    def get_version(self) -> str: ...
+    def get_plugins_info(self, depth: int = 2) -> Mapping[Any, Any]: ...
+    def get_plugin_info(self, name: str, depth: int = 2) -> Mapping[Any, Any]: ...
+    def get_plugins(self, depth: int = 2) -> Mapping[Any, Any]: ...
+    def get_jobs(
+        self, folder_depth: int = 0, folder_depth_per_request: int = 10, view_name: str | None = None
+    ) -> Sequence[Mapping[str, str]]: ...
+    def get_all_jobs(
+        self, folder_depth: int | None = None, folder_depth_per_request: int = 10
+    ) -> Sequence[Mapping[str, str]]: ...
+    def copy_job(self, from_name: str, to_name: str) -> None: ...
+    def rename_job(self, from_name: str, to_name: str) -> None: ...
+    def delete_job(self, name: str) -> None: ...
+    def enable_job(self, name: str) -> None: ...
+    def disable_job(self, name: str) -> None: ...
+    def set_next_build_number(self, name: str, number: int) -> None: ...
+    def job_exists(self, name: str) -> bool: ...
+    def jobs_count(self) -> int: ...
+    def assert_job_exists(self, name: str, exception_message: str = "job[%s] does not exist") -> None: ...
+    def create_folder(self, folder_name: str, ignore_failures: bool = False) -> None: ...
+    def upsert_job(self, name: str, config_xml: str) -> None: ...
+    def check_jenkinsfile_syntax(self, jenkinsfile: str) -> Sequence[str]: ...
+    def create_job(self, name: str, config_xml: str) -> None: ...
+    def get_job_config(self, name: str) -> str: ...
+    def reconfig_job(self, name: str, config_xml: str) -> None: ...
+    @overload
+    def build_job_url(
+        self, name: str, parameters: Mapping[str, Any] | Sequence[tuple[str, Any]] | None = None, token: Literal[""] | None = None
+    ) -> str: ...
+    @overload
+    def build_job_url(
+        self, name: str, parameters: dict[str, Any] | list[tuple[str, Any]] | None = None, token: str = ...
+    ) -> str: ...
+    @overload
+    def build_job(
+        self, name: str, parameters: Mapping[str, Any] | Sequence[tuple[str, Any]] | None = None, token: Literal[""] | None = None
+    ) -> int: ...
+    @overload
+    def build_job(self, name: str, parameters: dict[str, Any] | list[tuple[str, Any]] | None = None, token: str = ...) -> int: ...
```

Comment:
` = ...` would still allow matching this overload by omitting the parameter, which we don't want (it should only be matched when there's a value explicitly assigned.

The error `error: non-default argument follows default argument  [syntax]` can be fixed by splitting it into two overloads that match the two cases where this can happen.
```suggestion
    @overload
    def build_job_url(
        self, name: str, parameters: Mapping[str, Any] | Sequence[tuple[str, Any]] | None = None, token: Literal[""] | None = None
    ) -> str: ...
    @overload
    def build_job_url(self, name: str, parameters: dict[str, Any] | list[tuple[str, Any]] | None, token: str) -> str: ...
    @overload
    def build_job_url(
        self, name: str, parameters: dict[str, Any] | list[tuple[str, Any]] | None = None, *, token: str
    ) -> str: ...
    @overload
    def build_job(
        self, name: str, parameters: Mapping[str, Any] | Sequence[tuple[str, Any]] | None = None, token: Literal[""] | None = None
    ) -> int: ...
    @overload
    def build_job(self, name: str, parameters: dict[str, Any] | list[tuple[str, Any]] | None, token: str) -> int: ...
    @overload
    def build_job(self, name: str, parameters: dict[str, Any] | list[tuple[str, Any]] | None = None, *, token: str) -> int: ...
```

There's been some discussion to allow the previous form, but not much traction yet:
- https://discuss.python.org/t/some-improvements-to-function-overloads/36275/2
- https://mail.python.org/archives/list/python-dev@python.org/message/7W5RMJVBRN6NE6A3LUP5Y4BMOZKQRYWH/